app/models/announcement.rb

```diff
@@ -1,32 +1,31 @@
 class Announcement < ActiveRecord::Base
   attr_accessible :start_date, :end_date, :button, :link, :body
-  
+
   validates :button, presence: true
   validates :link, presence: true
   validates :body, presence: true
-  validates :start_date, presence: true, uniqueness: true
-  validates :end_date, presence: true, uniqueness: true
-  validate :overlapping_dates, on: :create
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :overlapping_dates
   validate :start_date_before_end_date
 
-  def date_range
-    start_date..end_date
-  end
-
   def overlapping_dates
-    announcements = Announcement.all
-    errors.add(:base, "Announcement conflicts with other announcements") if announcements.any? { |announcement| announcement.date_range.overlaps? date_range }
+    if start_date && end_date
+      Announcement.find do | d |
+        if (start_date <= d.end_date) && (end_date >= d.start_date) && (id != d.id)
+          errors.add(:base, "Conflicts with other Announcements " + d.start_date.to_s + ", " + d.end_date.to_s)
+        end
+      end
+    end
```

Comment:
It's finding all the Announcements and for each one, comparing it with the new/editted Announcement (local start_date/end_date to check for overlapping date. In the case of and edit/update it will not check for overlapping dates against itself since that is valid. It may be a bit heavier than possibly it needs to be but it is the most generalised case not knowing what the user might input for start_date and end_date. If a conflict is found, it shows the start_date and end_date of the conflict, otherwise the user would have to cancel their edits (which may be considerable) or save to a date which is known to be free. How does it look dangerous?